cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -396,67 +448,75 @@ def create_rna_dna_sample_case_map(self, rna_case: models.Family) -> Dict[str, D
+                f"Failed to link RNA samepl {rna_sample.internal_id} to dna samples - subject_id field is empty"
             )
 
-        subject_id_samples: Query = self.status_db.samples_by_subject_id(
-            customer_id=rna_sample.customer.internal_id,
-            subject_id=rna_sample.subject_id,
-            is_tumour=rna_sample.is_tumour,
-        )
-        nr_of_subject_id_dna_samples: int = len(subject_id_samples.all())
-        if nr_of_subject_id_dna_samples != 2:
-            raise CgDataError(
-                f"Failed to upload files for RNA case: unexpected number of DNA sample matches for subject_id: {rna_sample.subject_id}. Number of matches: {nr_of_subject_id_dna_samples} "
+        subject_id_samples: List[models.Sample] = self.validate_number_of_dna_samples_by_subject_id(
+            self.status_db.samples_by_subject_id(
+                customer_id=rna_sample.customer.internal_id,
+                subject_id=rna_sample.subject_id,
+                is_tumour=rna_sample.is_tumour,
             )
+        )
 
-        rna_dna_sample_case_map[rna_sample.internal_id]: Dict[str, list] = {}
+        rna_dna_sample_case_map[rna_sample.internal_id]: Dict[str, List[str]] = {}
         sample: models.Sample
         for sample in subject_id_samples:
-            if sample.internal_id == rna_sample.internal_id:
-                continue
-            rna_dna_sample_case_map[rna_sample.internal_id][sample.name]: list = []
-            return sample
+            if sample.internal_id != rna_sample.internal_id:
+                rna_dna_sample_case_map[rna_sample.internal_id][sample.name]: List[str] = []
+                return sample
+
+    def validate_number_of_dna_samples_by_subject_id(
+        self, subject_id_samples: List[models.Sample]
+    ) -> List[models.Sample]:
+        """Validates that there are two DNA samples with the same subject_id."""
+
+        if len(subject_id_samples) != 2:
+            raise CgDataError(
+                f"Unexpected number of DNA sample matches for subject_id: {subject_id_samples[0].subject_id}.\n"
+                f" Number of matches: {len(subject_id_samples)}"
+            )
+        return subject_id_samples
 
     @staticmethod
-    def _add_dna_cases_to_dna_sample(
+    def _map_dna_cases_to_dna_sample(
         dna_sample: models.Sample,
         rna_dna_sample_case_map: Dict[str, Dict[str, list]],
         rna_sample: models.Sample,
     ) -> None:
-        for dna_sample.link in dna_sample.links:
-            case_object: models.Family = dna_sample.link.family
+        """Maps a list of DNA cases linked to DNA sample."""
+        for link in dna_sample.links:
             if (
-                case_object.data_analysis in [Pipeline.MIP_DNA, Pipeline.BALSAMIC]
-                and case_object.customer_id == rna_sample.customer_id
+                link.family.data_analysis in [Pipeline.MIP_DNA, Pipeline.BALSAMIC]
```

Comment:
I wonder if `case.data_analysis` is not more intuitive (and shorter) than `link.family.data_analysis`